src/main/java/seedu/rc4hdb/ui/history/CommandHistory.java

```diff
@@ -0,0 +1,95 @@
+package seedu.rc4hdb.ui.history;
+
+import java.util.ArrayDeque;
+import java.util.Deque;
+
+import javafx.scene.input.KeyCode;
+
+/**
+ * The component that is responsible for logging and keeping track of user command inputs.
+ * There is no limitation to the number of commands it can keep track. The stack is cleared upon
+ * exiting the system.
+ */
+public class CommandHistory {
+    private static final String EMPTY_TEXT = "";
+
+    private Deque<String> forwardHistory = new ArrayDeque<String>();
+    private Deque<String> backwardHistory = new ArrayDeque<String>();
+
+    /**
+     * Adds the most recently executed command to the forwardHistory stack.
+     *
+     * @param commandText both valid and invalid commands are valid texts.
+     */
+    public void add(String commandText) {
+        assert(commandText != null);
```

Comment:
Good use of assertions in your code. Alternatively, you can use `requireNonNull` as well. 😄